packages/react-devtools-shared/src/hook.js

```diff
@@ -163,156 +178,147 @@ export function installHook(target: any): DevToolsHook | null {
-              let isInStrictMode = false;
-
-              // Search for the first renderer that has a current Fiber.
-              // We don't handle the edge case of stacks for more than one (e.g. interleaved renderers?)
-              const {getCurrentFiber, getIsStrictMode} = renderer;
-              if (typeof getCurrentFiber !== 'function') {
-                return;
-              }
-
-              const current: ?Fiber = getCurrentFiber();
-              if (current != null) {
-                try {
-                  if (
-                    typeof getIsStrictMode === 'function' &&
-                    getIsStrictMode()
-                  ) {
-                    isInStrictMode = true;
-                  }
-                } catch (error) {
-                  // Don't let a DevTools or React internal error interfere with logging.
-                }
-              }
-
-              if (isInStrictMode) {
-                if (!hideConsoleLogsInStrictMode) {
-                  // Dim the text color of the double logs if we're not
-                  // hiding them.
-                  let color;
-                  switch (method) {
-                    case 'warn':
-                      color =
-                        browserTheme === 'light'
-                          ? process.env.LIGHT_MODE_DIMMED_WARNING_COLOR
-                          : process.env.DARK_MODE_DIMMED_WARNING_COLOR;
-                      break;
-                    case 'error':
-                      color =
-                        browserTheme === 'light'
-                          ? process.env.LIGHT_MODE_DIMMED_ERROR_COLOR
-                          : process.env.DARK_MODE_DIMMED_ERROR_COLOR;
-                      break;
-                    case 'log':
-                    default:
-                      color =
-                        browserTheme === 'light'
-                          ? process.env.LIGHT_MODE_DIMMED_LOG_COLOR
-                          : process.env.DARK_MODE_DIMMED_LOG_COLOR;
-                      break;
-                  }
-
-                  if (color) {
-                    originalMethod(`%c${format(...args)}`, `color: ${color}`);
-                  } else {
-                    throw Error('Console color is not defined');
-                  }
-                }
-              } else {
-                originalMethod(...args);
-              }
-            };
-
-            overrideMethod.__REACT_DEVTOOLS_ORIGINAL_METHOD__ = originalMethod;
-            originalMethod.__REACT_DEVTOOLS_OVERRIDE_METHOD__ = overrideMethod;
-
-            // $FlowFixMe property error|warn is not writable.
-            targetConsole[method] = overrideMethod;
-          } catch (error) {}
-        });
+    }: {
+      hideConsoleLogsInStrictMode: boolean,
+      browserTheme: BrowserTheme,
+    },
+  ) {
+    const overrideConsoleMethods = ['error', 'trace', 'warn', 'log'];
+
+    if (unpatchFn !== null) {
```

Comment:
Since we only patch once, do we need to use a ref-like object to store `browserTheme` and `hideConsoleLogsInStrictMode` preferences (like we do in `patch` in `packages/react-devtools-shared/src/backend/console`?)

Reply:
I'm not sure what you mean. We're currently not using a ref-like object to store these preferences because we should only call these functions before DevTools finishes its handshake (ie during the initial render in the extension)

Reply:
We talked about this outside of the PR and you're right. It shouldn't be necessary in this case. 👍🏼